File: determinant_inverse_transpose.py
import numpy as np

def transpose(matrix):
    new_matrix = []
    for i in range(len(matrix[0])):
        row = []
        for j in range(len(matrix)):
            row.append(matrix[j][i])
        new_matrix.append(row)
    return new_matrix

def determinant(matrix):
    if len(matrix) == 1:
        return matrix[0,0]
    else:
        determinant_number = 0
        for i in range(len(matrix)):
            determinant_number += matrix[0,i] * (-1)**(i+1) * determinant(slice_matrix(matrix,i))
    return determinant_number

def slice_matrix(matrix,i):
    new_matrix=[]
    for j in range(len(matrix)):
        if j!=i:
            new_matrix.append(matrix[1:,j])
    return np.array(new_matrix)

def slice_matrix_2D(matrix,i,j):
    pos_r = 0
    pos_c = 0
    new_matrix = np.empty((matrix.shape[0]-1,matrix.shape[1]-1))
    for k in range(matrix.shape[0]):
        for m in range(matrix.shape[1]):
            if k!=i and m!=j:
                new_matrix[pos_r,pos_c] = matrix[k,m]
                if pos_c < new_matrix.shape[1]-1:
                    pos_c += 1
                else:
                    pos_c = 0
                    pos_r += 1
    return new_matrix

def adjugate(matrix):
    adjugate_matrix=np.copy(matrix)
    for i in range(adjugate_matrix.shape[0]):
        for j in range(adjugate_matrix.shape[1]):
            adjugate_matrix[i,j] = determinant(slice_matrix_2D(matrix,i,j))*(-1)**(i+j+3)
    adjugate_matrix = transpose(adjugate_matrix)
    return adjugate_matrix

    
def inverse(matrix):
    inverse_matrix = adjugate(matrix)/determinant(matrix)
    return inverse_matrix

File: determinantInverseTranspose.js
const { simplify } = require('mathjs');

var transpose = function(matrix) {
    var rows = matrix.length;
    var columns = matrix[0].length;
    var res = Array(columns);
    res.fill(Array(rows).fill(0));
    for(var i = 0; i < rows; i++)
    {
        for(var j = 0; j < columns; j++)
        {
            res[j][i] = matrix[i][j];
        }
    }
    return res;
}

var determinant = function(matrix) {
    if (matrix.length == 1)
    {
        return matrix[0][0];
    }
    else
    {
        var determinantNumber = 0;
        for(var i = 0; i < matrix.length; i++)
        {
            var temp = multiply(matrix[0][i], pow(-1, i + 1));
            determinantNumber = add(determinantNumber, multiply(temp, determinant(sliceMatrix(matrix, i))));
        }
    }
    return determinantNumber;
}

var sliceMatrix = function(matrix, index) {
    var newMatrix = [];
    var tempMatrix = [];
    for(var i = 0; i < matrix.length; i++)
    {
        if(i != index)
        {
            newMatrix.push(matrix[i].slice(0, index).concat(matrix[i].slice(index + 1)));
        }
    }
    return newMatrix;
}

var adjugate = function(matrix) {
    for(var i = 0; i < matrix.lenght; i++)
    {
        for(var j = 0; j < matrix.length; j++)
        {
            matrix[i][j] = multiply(determinant(sliceMatrix2D(matrix, i, j)), pow(-1, i + j + 3));
        }
    }
    matrix = transpose(matrix);
    return matrix;
}

var sliceMatrix2D = function(matrix, i, j) {
    var pos_r = 0;
    var pos_c = 0;
    var newMatrix = Array(matrix.length - 1);
    newMatrix.fill(Array(matrix[0].length - 1));
    for(var r = 0; r < matrix.length; r++)
    {
        for(var s = 0; s < matrix[0].length; s++)
        {
            if(r != i && s != j)
            {
                newMatrix[pos_r][pos_c] = matrix[r][s];
                if (pos_c < newMatrix[0].lenght - 1)
                {
                    pos_c++;
                }
                else
                {
                    pos_c = 0;
                    pos_r++;
                }
            }
        }
    }
}

var inverse = function(matrix) {
    inverseMatrix = divide(adjugate(matrix), determinant(matrix));
    return inverseMatrix;
}